### PR TITLE
Contamination & Visibility : `MIRI` expansion

### PR DESCRIPTION
**Pull request by jaymedina**
This will be the PR for the expansion of contamination & visibility tool for MIRI LRS observations, and an update to NIRCam.

What's left:
- [x] Fix pileup issue with MIRI
- [x] Verify that contamination plot for MIRI is correct (i.e. no missing contamination)
- [x] Flip the MIRI trace to make sure orientation of contamination plot is correct 
- [x] Verify that NIRCam trace is oriented correctly 
- ~Generate lamO1.txt lamO2.txt files for NIRCam and MIRI~ Not useful at the moment. Perhaps in another update.
- [x] Put in the `if lamO1 < 1.X: continue` piece of code for order 1
- [x] Change colorbar to sets of colors that are easier to see (maybe light blue to dark blue to violet)

What's left (08-25-2020): 
- [x] [NIRCam] Add updated code to gtsFieldSim()
- [x] [NIRCam] Test gtsFieldSim() with fieldSim-verify notebook 
- [x] Wrap up fieldSim-verify notebook
- [x] [MIRI] contamFig update
- ~Add contamination results to downloadable CSV file~ Not yet
- [ ] Make it pep8

### CODE REVIEW

File: exoctk/contam_visibility/miniTools.py
Comment (bourque) [inline exoctk/contam_visibility/miniTools.py:15]: I think PEP8 prefers to list imports in alphabetical order (in each import section)?

File: exoctk/contam_visibility/miniTools.py
Comment (bourque) [inline exoctk/contam_visibility/miniTools.py:56]: I think PEP8 prefers variable names to use underscores (i.e. `all_ra`) instead of CamelCase

File: exoctk/contam_visibility/miniTools.py
Comment (bourque) [inline exoctk/contam_visibility/miniTools.py:85]: PEP8 prefers that variable names be lowercase unless they are global variables.  But `FILE` doesn't appear to be a global variable here?

File: exoctk/contam_visibility/miniTools.py
Comment (bourque) [inline exoctk/contam_visibility/miniTools.py:42]: I think PEP8 prefers that function names use underscores (e.g. `plot_temps`) instead of CamelCase.  Usually CamelCase is used for class definitions.

File: exoctk/contam_visibility/miniTools.py
Comment (bourque) [inline exoctk/contam_visibility/miniTools.py:106]: There seems to be a lot of uppercase variable names here even though they are not global variables?

File: exoctk/contam_visibility/sossContamFig.py
Comment (bourque) [inline exoctk/contam_visibility/sossContamFig.py:73]: I would suggest using more explicit variable names (e.g. `trace` instead of `tr`), as variable names with one or two characters are sometimes hard to read.  I think PEP8 technically outlaws single-character variable names.

File: exoctk/exoctk_app/app_exoctk.py
Comment (bourque) [inline exoctk/exoctk_app/app_exoctk.py:532]: Probably best to clean up commented out bits if they are not going to be used in production

**Review by bourque — APPROVED** [review]
@jaymedina Amazing work getting all of these changes in, there are a lot of them!  From a software perspective, this looks good.  I only have PEP-8 related nit-picky suggestions, namely:

- Variable names should be lowercase_underscore
- Function names should be lowercase_underscore
- Only global variable names should be uppercase
- Explicit variable names (e.g. `instrument`) are better than unnecessarily-abbreviated variable names (e.g. `inst`)
- import sections should be in alphabetical order

I don't view these are critical for the software to run, so I would deem this PR 'approved'; I will leave it up to you if you would like to go in and make those PEP-8 changes.  

Do any sphinx/API docs or unit tests need to be updated or added for this PR?

I don't posses enough domain expertise to say if the changes are good from a science perspective 🤷  It looks like some really great improvements though!

**Comment by jaymedina** [comment]
Thanks for the review @bourque 👍 

> Do any sphinx/API docs or unit tests need to be updated or added for this PR?

We'll need to add a unit test for `miniTools.py` as the Contam Verify mini tool is in there. Is sphinx/API docs referring to our Read The Docs pages?

**Comment by bourque** [comment]
@jaymedina 

> Is sphinx/API docs referring to our Read The Docs pages?

Yeah exactly.  The part of the RtD that renders the docstrings.

**Comment by jaymedina** [comment]
> @jaymedina
> 
> > Is sphinx/API docs referring to our Read The Docs pages?
> 
> Yeah exactly. The part of the RtD that renders the docstrings.

Yes, the sphinx/API should probably have the docstrings for miniTools.py and a unit test should be added as well.

**Comment by jaymedina** [comment]
I'm going to run autopep8 on all the files, just to patch things up a bit, and will merge tonight.

**Pull request merged**
